analytic_engine/src/instance/serial_executor.rs

```diff
@@ -131,7 +131,10 @@ impl TableFlushScheduler {
                     }
                     FlushState::Flushing => (),
                     FlushState::Failed { err_msg } => {
-                        return BackgroundFlushFailed { msg: err_msg }.fail();
+                        warn!("Re-flush memory tables after background flush failed:{err_msg}");
+                        // Mark the worker is flushing.
+                        *flush_state = FlushState::Flushing;
```

Comment:
Maybe add an option?
For some storage, failure maybe unrecoverable and retry may be meaningless?

Reply:
I guess a better way is to support limited retry, that is to say, we can use a counter to record the times of continuous flush failures (the counter will be reset to 0 once a flush succeeds). If the counter reaches a limit, we can set the `flush_state` as `FlushState::Failed`.